frontend/app/components/resizer/main-menu-resizer.component.ts

```diff
@@ -0,0 +1,180 @@
+//-- copyright
+// OpenProject is a project management system.
+// Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License version 3.
+//
+// OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+// Copyright (C) 2006-2013 Jean-Philippe Lang
+// Copyright (C) 2010-2013 the ChiliProject Team
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+//
+// See doc/COPYRIGHT.rdoc for more details.
+//++
+
+import {Component, ElementRef, HostListener, Injector, Input, OnDestroy, OnInit} from '@angular/core';
+
+@Component({
+  selector: 'main-menu-resizer',
+  template: `
+    <div class="main-menu--resizer" ng-class="{ 'show': !showNavigation }">
+      <a href="#"
+         title="{{toggleTitle}}"
+         class="main-menu--navigation-toggler"
+         ng-click="mainMenu.toggleNavigation($event)">
+        <i class="icon-resizer-vertical-lines" aria-hidden="true"></i>
+      </a>
+    </div>
+  `
+})
+
+export class MainMenuResizerDirective implements OnInit, OnDestroy {
+  private elementClass:string;
+  private resizeEvent:string;
+  private localStorageKey:string;
+
+  private resizingElement:HTMLElement;
+  private elementWidth:number;
+  private oldPosition:number;
+  private mouseMoveHandler:any;
+  private element:HTMLElement;
+  private htmlNode:HTMLElement;
+  private toggleTitle:string;
```

Comment:
Properties in the view should be public, not private. This is irrelevant during runtime since these modifiers are only for type-checking. But they will raise errors when AOT is enabled.